### ISSUE TITLE
ReadWriteMany slow and not required

### ISSUE THREAD

**Issue by wernight** [bitwarden/helm-charts#150]
### Steps To Reproduce

Set values `general.volumeAccessMode=ReadWriteMany` and `sharedStorageClassName=standard` and deploy.

### Expected Result

Should work and work faster & cheaper than NFS.

### Actual Result

Without any changes this fails with a "Multi-Attach error for volume" error because different Pods try to access the same PVC. For example `dataprotection`, `licences`, `applogs` and `secrets-store-inline` are shared among 4 or 5 Pods.

### Additional Context

The trick to avoid **ReadWriteMany** is using *ReadWriteOnce* which allows multiple Pods on the same Node to use a volume (usually). A more crude alternative is using a single Pod with multiple Containers. Also using *ReadOnlyMany* for any place that doesn't require writing.

Note: I think this is a bug because I see no reason common alternatives above wouldn't work in this case.

### Chart Version

self-host-2024.5.0

### Issue Tracking Info

- [ ] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by gecube** [comment]
I totally agree, I think that RWX is overkill and it could be sufficient to use RWO volumes. Unfortunately, it would need massive change in the chart.

**Comment by gecube** [comment]
another good option could be to replace RWX volume with direct S3 support in the application. And it will be much more scalable than using EFS on EKS.

**Comment by keithhubner** [comment]
Hi, thank you for raising this. 

RWX is required by the application due to the way containers read/write to storage, for example attachment storage and data protection keys. Changing this behavior will require development effort.
We have introduced RWO for MSSQL in this PR: https://github.com/bitwarden/helm-charts/pull/447.

This is therefore not a bug but a feature request and we will ensure this is raised with the relevant team(s).

Thank you again for raising this.
